### PR TITLE
[WIP] build-style/meson: fix building rust apps

### PR DESCRIPTION
**Pull request by Cogitri**

### CODE REVIEW

**Comment by Cogitri** [comment]
Will test gnome-podcasts tomorrow.

File: common/build-style/meson.sh [hunk 1]
```diff
@@ -64,6 +65,17 @@ endian = '${_MESON_TARGET_ENDIAN}'
 EOF
 		configure_args+=" --cross-file=${meson_crossfile}"
 
+		# Fix building Rust apps
+		mkdir .cargo
```
Comment (jnbr) [inline common/build-style/meson.sh]: better use `mkdir -p .cargo` here

Comment (Cogitri) [reply common/build-style/meson.sh]: How could ${wrksrc} not exist here?

Comment (jnbr) [reply common/build-style/meson.sh]: `.cargo` could exist already

Comment (Cogitri) [reply common/build-style/meson.sh]: Oh, right, -p also deals with that

File: common/environment/build-style/meson.sh [hunk 2]
```diff
@@ -1 +1,9 @@
 hostmakedepends+=" meson"
+
+export PKG_CONFIG_ALLOW_CROSS=1
+
+export LIBGIT2_SYS_USE_PKG_CONFIG=1
+export GETTEXT_BIN_DIR=/usr/bin
+export GETTEXT_LIB_DIR="${XBPS_CROSS_BASE}/usr/lib/gettext"
+export GETTEXT_INCLUDE_DIR="${XBPS_CROSS_BASE}/usr/include"
+export LIBSSH2_SYS_USE_PKG_CONFIG=1
```
Comment (maxice8) [inline common/environment/build-style/meson.sh]: missing newline here

Comment (jnbr) [reply common/environment/build-style/meson.sh]: I would prefer a solution which doesn't involve maintaining two copies of this list

Comment (Cogitri) [reply common/environment/build-style/meson.sh]: Hm, can we share envs between build styles? Adding it to the global env seems meh to me

**Comment by maxice8** [comment]
On Thu, Jan 17, 2019 at 04:44:37PM -0800, Rasmus Thomsen wrote:
> Cogitri commented on this pull request.
> 
> 
> 
> > @@ -1 +1,9 @@
>  hostmakedepends+=" meson"
> +
> +export PKG_CONFIG_ALLOW_CROSS=1
> +
> +export LIBGIT2_SYS_USE_PKG_CONFIG=1
> +export GETTEXT_BIN_DIR=/usr/bin
> +export GETTEXT_LIB_DIR="${XBPS_CROSS_BASE}/usr/lib/gettext"
> +export GETTEXT_INCLUDE_DIR="${XBPS_CROSS_BASE}/usr/include"
> +export LIBSSH2_SYS_USE_PKG_CONFIG=1
> 
> Hm, can we share envs between build styles? Adding it to the global env seems meh to me
> 
> -- 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/7341#discussion_r248893960

NOt between build_styles, maybe we can change that.

File: common/environment/build-style/meson.sh (hunk 2 above)
Comment (maxice8) [reply common/environment/build-style/meson.sh]: @jnbr #7372

**Comment by maxice8** [comment]
This will be fixed by build_helper #7505

**Pull request closed without merging**